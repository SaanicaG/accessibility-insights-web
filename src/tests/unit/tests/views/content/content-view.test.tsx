// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { render } from '@testing-library/react';
import { NarrowModeDetector } from 'DetailsView/components/narrow-mode-detector';
import * as React from 'react';
import { mockReactComponents } from 'tests/unit/mock-helpers/mock-module-helpers';

import { ContentView, ContentViewDeps } from 'views/content/content-view';
jest.mock('DetailsView/components/narrow-mode-detector');
describe('content view', () => {
    mockReactComponents([NarrowModeDetector]);

    it('renders', () => {
        const applicationTitle = 'THE_APPLICATION_TITLE';
        const deps = {
            textContent: {
                applicationTitle,
            },
        } as ContentViewDeps;

        const component = (
            <ContentView deps={deps}>
                <p>THE CONTENT</p>
            </ContentView>
        );

        const renderResult = render(component);

        expect(renderResult.asFragment()).toMatchSnapshot();
    });
});
