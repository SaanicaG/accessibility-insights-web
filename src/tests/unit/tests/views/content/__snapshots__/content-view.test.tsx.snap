// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content view renders 1`] = `
<DocumentFragment>
  <mock-page
    deps="[object Object]"
  >
    <div
      class="content-container"
    >
      <div
        class="content-left"
      />
      <div
        class="content"
      >
        <p>
          THE CONTENT
        </p>
      </div>
      <div
        class="content-right"
      />
    </div>
  </mock-page>
</DocumentFragment>
`;
